src/lib/sql/abstract_cache.hpp

```diff
@@ -36,6 +36,9 @@ class AbstractCache {
   // Remove all elements from the cache and resize to the given capacity.
   virtual void clear_and_resize(size_t capacity) = 0;
 
+  // Resize to the given capacity.
+  virtual void resize(size_t capacity) = 0;
```

Comment:
Now that we have `resize`, what it `clear_and_resize`'s reason to exist?

Reply:
Good point. We could either remove it completely or implement it in the AbstractCache by calling `clear()` and `resize()`

Reply:
The `AbstractCache` already has `clear()`. As no one uses `clear_and_resize` now and because it can be easily replaced with clear+resize, let's get rid of it.

Reply:
Done